### DISCUSSION TITLE
TMSU integration

### DISCUSSION THREAD

**Discussion by nkh** [cmus/cmus#1088] [General]
Hej, new to cmus

I wonder if you've seen an integration to tmsu somewhere. I'd write one but I haven't found a way to get the list of files selected, I know I can get the current  track with -Q.

**Comment by jayay** [comment]
Hi,

I assume you're asking a question here. Could you please be more precise about what you are up to and what you need?

**Comment by nkh** [comment]
Hi,

I want to tag files in tmsu (but maybe also add mp3 tags).

use case 1, in another terminal using fzf:
===============================

-get cmus selection
-get tags from user 
-tag files

that I got on the way , I had to write a small C program (get_cmus_selection) as I couldn't get the list of selected files via  cmus-remote -C 'run ...' with a bash script (works on my box as it uses other scripts of mine)

#!/bin/bash -i

selected_files="$(pwd)/selected_files"

cmus-remote -C "set confirm_run=0"
cmus-remote -C "run get_cmus_selection $selected_files"

tags=$(tmsu tags | \
	fzf -e -m --info=inline --ansi --color="hl:51" --no-sort --layout=reverse-list \
		--bind='alt-k:preview-up' --bind='alt-j:preview-down' \
		--bind='?:toggle-preview' \
		--preview-window='top:50%' \
		--preview "/bin/cat $selected_files | shorten -l -e … " \
	)

set -e
[[ -z "$tags" ]] && false

tags=$(echo -e "$tags" | tr '\n' ' ')
ocat "$selected_files" | tr '\n' '\00' || xargs -0 -n 1 -i echo tmsu tag $(echo \'{}\')  $tags



use case 2, in cmus' command :
========================

something like
: tmsu tag1 tag2

it would be cool if one could also pop up an fzf like selection window
